boards/arm/sparkfun_pro_micro_rp2040/board.cmake

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# This configuration allows selecting what debug adapter debugging to use
+# for the SparkFun Pro Micro RP2040 by a command-line argument.
+# It is mainly intended to support both the 'picoprobe' and 'raspberrypi-swd'
+# adapter described in "Getting started with Raspberry Pi Pico".
+# And any other SWD debug adapter might also be usable with this configuration.
+
+# Note that the Pro Micro RP2040 SWD pins are test points on the underside
+# of the board.
+
+# Set RPI_PICO_DEBUG_ADAPTER to select debug adapter by command-line arguments.
+# e.g.) west build -b rpi_pico -- -DRPI_PICO_DEBUG_ADAPTER=raspberrypi-swd
+# The value is treated as a part of an interface file name that
+# the debugger's configuration file.
+# The value must be the 'stem' part of the name of one of the files
+# in the openocd interface configuration file.
+# The setting is store to CMakeCache.txt.
+if ("${RPI_PICO_DEBUG_ADAPTER}" STREQUAL "")
+	set(RPI_PICO_DEBUG_ADAPTER "cmsis-dap")
+endif()
+
+board_runner_args(openocd --cmd-pre-init "source [find interface/${RPI_PICO_DEBUG_ADAPTER}.cfg]")
+board_runner_args(openocd --cmd-pre-init "transport select swd")
+board_runner_args(openocd --cmd-pre-init "source [find target/rp2040.cfg]")
+
+# The adapter speed is expected to be set by interface configuration.
+# But if not so, set 2000 to adapter speed.
+board_runner_args(openocd --cmd-pre-init "set_adapter_speed_if_not_set 2000")
+
+board_runner_args(jlink "--device=RP2040_M0_0")
+board_runner_args(uf2 "--board-id=RPI-RP2")
+board_runner_args(pyocd "--target=rp2040")
+
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
```

Comment:
think uf2 should come first in the list so it is used first, most people will be programming this using the uf2 bootloader

Reply:
Isn't the default set above in the little if block? I can change that one to UF2. I'm not sure if inclusion order matters.

Reply:
Ok, I'm pretty sure order doesn't matter. It always builds UF2, regardless of debug settings, as that it set by the defconfig.  And UF2 isn't a debug adaptor. I just copied this from all of the other rp2040 boards that are supported.

Reply:
> Ok, I'm pretty sure order doesn't matter. 

No it isn't.
We determine the default debug/flash setting with the order of runner, or following the setting of `board_set_runner_ifnset`/`board_set_runner_ifnset` if these are called.
And the runner setting know itself support flashing or debugging or both.
Since the debug terminals on this board are particularly difficult to use for usual, it seems easier to use settings that prioritize flash using uf2.